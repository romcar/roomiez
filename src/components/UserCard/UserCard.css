.profilePic {
	margin: 15px 0 15px 25px;
	border-radius: 75px;
	width: 115px;
	border: 5px solid #ff8484; /* pink outline */
}
