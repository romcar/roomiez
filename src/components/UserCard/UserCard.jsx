import React from 'react';
import './UserCard.css';
import { Container, Col, Row, CardImg } from 'reactstrap';

const UserCard = () => (
	<div>
		<Container>
			<Row>
				<Col xl={2.25} md={2.25} sm={2.25}>
					<CardImg className="profile-pic" src="http://placekitten.com/120/120" />
				</Col>
				<Col>
					<h4 className="user-name">user name</h4>
					<div className="chill-lvl">chill level</div>
					<div className="questions-answered"># qustions</div>
					<div className="isMoving">isMoving button</div>
				</Col>
			</Row>
		</Container>
	</div>
);

export default UserCard;
