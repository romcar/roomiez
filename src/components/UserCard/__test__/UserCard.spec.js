/* esling  */
import React from 'react';
import { configure, shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import UserCard from '../UserCard';
// import the component

configure({ adapter: new Adapter() });
describe('<UserCard/>', () => {
	it('Should render correctly', () => {
		const component = shallow(<UserCard />); // render with no attributes

		expect(toJSON(component)).toMatchSnapshot(); // 1st time is always right
	});

	it("Should display the user's name", () => {
		const component = shallow(<UserCard />);
		expect(component.find('.user-name').length).toEqual(1);
	});

	it("Should display the user's chill level", () => {
		const component = shallow(<UserCard />);
		expect(component.find('.chill-lvl').length).toEqual(1);
	});

	it('Should have somewhere to put the number of questions answered', () => {
		const component = shallow(<UserCard />);
		expect(component.find('.questions-answered').length).toEqual(1);
	});

	it('Should have somewhere to place a isMoving switch', () => {
		const component = shallow(<UserCard />);
		expect(component.find('.isMoving').length).toEqual(1);
	});
});
