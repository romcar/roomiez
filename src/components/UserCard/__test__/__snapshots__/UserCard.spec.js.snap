// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard/> Should render correctly 1`] = `
<div>
  <Container
    tag="div"
  >
    <Row
      tag="div"
    >
      <Col
        md={2.25}
        sm={2.25}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xl={2.25}
      >
        <CardImg
          src="http://placekitten.com/120/120"
          tag="img"
        />
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <h4
          className="user-name"
        >
          user name
        </h4>
        <div
          className="chill-lvl"
        >
          chill level
        </div>
        <div
          className="questions-answered"
        >
          # qustions
        </div>
        <div
          className="isMoving"
        >
          isMoving button
        </div>
      </Col>
    </Row>
  </Container>
</div>
`;
